notebooks/niriss_ami_binary/2_niriss_ami_binary.ipynb

```diff
@@ -11,19 +11,21 @@
    "cell_type": "markdown",
```

Comment:
<ol><li><code>boxfile</code> and other variables that refer to paths should use <code>pathlib.Path()</code> objects instead of strings for better compatibility across operating systems. You would need to add <code>from pathlib import Path</code> to the main import cell and then can change to <code>boxfile = Path('./niriss_ami_binary2_inflight.zip')</code> in this cell. </li><li>The call to <code>os.system()</code> at the end of the cell is considered unsafe and should be removed. It may not even be necessary to move these files; we would just need to account for their different location</li></ol><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/spacetelescope/jdat_notebooks/pull/99/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Reply:
I remember having a problem running the notebook without moving the MASK files to the working directory and at the time I could not find a way to work around it. I will try again.

Reply:
I also tried changing 'outdir = './pipeline_calibrated_data/'' to use Path, but the pipeline did not like it.

Reply:
Same for 'bpcorrdir = './pipeline_calibrated_data_corr/''

Reply:
Ok, problem.
from run_bp_fix import correct_fitsfiles requires the MASK file to be already available, but we download it in the following cell. I can change the path to the MASK file in run_bp_fix so we do not need to move it to the working directory, but this does not solve the problem that the file is required at import. Is there a best way to proceed here?

Reply:
Did you try following the instructions from the first item in [this comment](https://github.com/spacetelescope/jdat_notebooks/pull/99#issuecomment-1322399009)? It should resolve the import issue.

If the pipeline doesn't like `Path` objects, possible solutions are to use `str(path1)` or `path1.as_posix()` to convert the object to a string when you want to include `path1` as an argument to pipeline-related function.

Reply:
Either of those solutions should allow you to use `Path()` in the places where you said it wasn't working in this thread and others. Since paths on some operating systems `work/like/this` and others `work\like\that`, routing the path strings through `Path()` first helps make sure they're rendered correctly for a given user's OS.

Reply:
Whether it is commented or not, I get:

---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
Input In [2], in <cell line: 19>()
     16 from nrm_analysis import nrm_core, InstrumentData
     17 from nrm_analysis.misctools.implane2oifits import calibrate_oifits
---> 19 from run_bp_fix import correct_fitsfiles

File ~/Documents/JDAT/GITDUMP/niriss_ami_binary/run_bp_fix.py:54, in <module>
     52 pupilfile_nrm = "./niriss_ami_binary2_inflight/MASK_NRM.fits"
     53 pupilfile_clearp = "./niriss_ami_binary2_inflight/MASK_CLEARP.fits"
---> 54 nrm_pupil = pyfits.getdata(pupilfile_nrm)
     55 clearp_pupil = pyfits.getdata(pupilfile_clearp)
     56 pupil_masks = {
     57     "NRM": nrm_pupil,
     58     "CLEARP": clearp_pupil,
     59 }

File ~/opt/miniconda3/envs/jdatnb/lib/python3.8/site-packages/astropy/io/fits/convenience.py:203, in getdata(filename, header, lower, upper, view, *args, **kwargs)
    199 extver = kwargs.get('extver')
    200 ext_given = not (len(args) == 0 and ext is None and
    201                  extname is None and extver is None)
--> 203 hdulist, extidx = _getext(filename, mode, *args, **kwargs)
    204 try:
    205     hdu = hdulist[extidx]

File ~/opt/miniconda3/envs/jdatnb/lib/python3.8/site-packages/astropy/io/fits/convenience.py:1071, in _getext(filename, mode, ext, extname, extver, *args, **kwargs)
   1068 elif extver and extname is None:
   1069     raise TypeError('extver alone cannot specify an extension.')
-> 1071 hdulist = fitsopen(filename, mode=mode, **kwargs)
   1073 return hdulist, ext

File ~/opt/miniconda3/envs/jdatnb/lib/python3.8/site-packages/astropy/io/fits/hdu/hdulist.py:175, in fitsopen(name, mode, memmap, save_backup, cache, lazy_load_hdus, ignore_missing_simple, **kwargs)
    172 if not name:
    173     raise ValueError(f'Empty filename: {name!r}')
--> 175 return HDUList.fromfile(name, mode, memmap, save_backup, cache,
    176                         lazy_load_hdus, ignore_missing_simple, **kwargs)

File ~/opt/miniconda3/envs/jdatnb/lib/python3.8/site-packages/astropy/io/fits/hdu/hdulist.py:410, in HDUList.fromfile(cls, fileobj, mode, memmap, save_backup, cache, lazy_load_hdus, ignore_missing_simple, **kwargs)
    398 @classmethod
    399 def fromfile(cls, fileobj, mode=None, memmap=None,
    400              save_backup=False, cache=True, lazy_load_hdus=True,
    401              ignore_missing_simple=False, **kwargs):
    402     """
    403     Creates an `HDUList` instance from a file-like object.
    404 
   (...)
    407     documentation for details of the parameters accepted by this method).
    408     """
--> 410     return cls._readfrom(fileobj=fileobj, mode=mode, memmap=memmap,
    411                          save_backup=save_backup, cache=cache,
    412                          ignore_missing_simple=ignore_missing_simple,
    413                          lazy_load_hdus=lazy_load_hdus, **kwargs)

File ~/opt/miniconda3/envs/jdatnb/lib/python3.8/site-packages/astropy/io/fits/hdu/hdulist.py:1060, in HDUList._readfrom(cls, fileobj, data, mode, memmap, cache, lazy_load_hdus, ignore_missing_simple, **kwargs)
   1057 if fileobj is not None:
   1058     if not isinstance(fileobj, _File):
   1059         # instantiate a FITS file object (ffo)
-> 1060         fileobj = _File(fileobj, mode=mode, memmap=memmap, cache=cache)
   1061     # The Astropy mode is determined by the _File initializer if the
   1062     # supplied mode was None
   1063     mode = fileobj.mode

File ~/opt/miniconda3/envs/jdatnb/lib/python3.8/site-packages/astropy/io/fits/file.py:170, in _File.__init__(self, fileobj, mode, memmap, overwrite, cache)
    168     self._open_fileobj(fileobj, mode, overwrite)
    169 elif isinstance(fileobj, (str, bytes)):
--> 170     self._open_filename(fileobj, mode, overwrite)
    171 else:
    172     self._open_filelike(fileobj, mode, overwrite)

File ~/opt/miniconda3/envs/jdatnb/lib/python3.8/site-packages/astropy/io/fits/file.py:562, in _File._open_filename(self, filename, mode, overwrite)
    559 ext = os.path.splitext(self.name)[1]
    561 if not self._try_read_compressed(self.name, magic, mode, ext=ext):
--> 562     self._file = open(self.name, IO_FITS_MODES[mode])
    563     self.close_on_error = True
    565 # Make certain we're back at the beginning of the file
    566 # BZ2File does not support seek when the file is open for writing, but
    567 # when opening a file for write, bz2.BZ2File always truncates anyway.

FileNotFoundError: [Errno 2] No such file or directory: './niriss_ami_binary2_inflight/MASK_NRM.fits'

Reply:
I'm not getting the same behavior as you with `run_bp_fix`, but I'll investigate further.

Reply:
The problems are the statements that define `nrm_pupil` and `clearp_pupil` in `run_bp_fix.py`. The rough explanation is that any time you import the script, Python tries to run everything that isn't contained in a function.

Those two objects are used to create `pupil_masks`, which is only used in the `calcpsf()` function. One solution would be to delete [lines 52 through 59](https://github.com/spacetelescope/jdat_notebooks/blob/a3df8d43462ad1702f2ff1603acc90d3e8ddf5f9/notebooks/niriss_ami_binary/run_bp_fix.py#L52:L59) and then replace the definition of `pupil_mask` in `calcpsf()` with the following:

```
if pupil == "NRM":
        pupil_mask = pyfits.getdata("MASK_NRM.fits")
    elif pupil == "CLEARP":
        pupil_mask = pyfits.getdata("MASK_CLEARP.fits")
    else:
        raise ValueError("pupil should be 'NRM' or 'CLEARP'")
```

Since `calcpsf()` isn't imported in the notebook, an easier but more drastic solution is to delete all of lines 52-59 *and* `calcpsf()` from the script, but I think the first option should be doable.

Reply:
Whichever route you choose, you should be able to leave the `if __name__ == 'main'` section uncommented like it was before.